### PR TITLE
Disabling the button in Stripe Elements flow

### PR DESCRIPTION
**Pull request by svillafe**
## Why are you doing this?

To disable stripe inline button after the first submit. 

## Changes

* Changing the button dynamically after the users click.
* Created the disabled class.

## Why are not you using redux to store the disabled state?

 In the beginning, I implemented in that way, but the fact that the state change and then re-render the component, it interacts with stripe-react-elements causing an error. Therefore, I am changing the "state" of the component without saving the change in the state and as a result, avoiding a re-render.

## Screenshots

disable

### CODE REVIEW

File: assets/components/stripeInlineForm/stripeInlineForm.jsx [hunk 1]
```diff
@@ -31,6 +32,9 @@ type PropTypes = {|
   isPostDeploymentTestUser: boolean,
   setError: (message: string) => void,
   resetError: () => void,
+  isSubmitButtonDisable: boolean,
```
Comment (joelochlann) [inline assets/components/stripeInlineForm/stripeInlineForm.jsx]: should be `Disabled`

**Comment by Ap0c** [comment]
@svillafe @joelochlann are you still working on this PR?

**Comment by joelochlann** [comment]
Yes, we've decided to keep the inline flow (as the normal flow, no longer a test) - need to speak to @svillafe today about getting this PR out

**Comment by joelochlann** [comment]
Santiago's at a conference but he'll look at this on Monday with help from @jranks123

**Comment by svillafe** [comment]
@joelochlann @Ap0c @jranks123  I will create a new PR to set up the inline stripe flow as the new control. Additionally, I will create a card in the top of the health backlog to fix the stripe test after we merge that PR.

**Comment by svillafe** [comment]
Closing this PR. The new PR is https://github.com/guardian/support-frontend/pull/784, I will be updating it and then I will let you know to review it. Thank you all!

**Comment by joelochlann** [comment]
> Additionally, I will create a card in the top of the health backlog to fix the stripe test after we merge that PR.

Do you mean you'll make the inline element the new control, and then fix the spinner/disabled issue in a subsequent PR?

**Comment by svillafe** [comment]
@joelochlann Sorry that was not clear enough. In #784 I was thinking to disable the button after submission and to apply it as a new control. Then in a following PR we will need to fix the post deploy test for Stripe, which will be broken from #784. I will put a card in the health backlog for that.

**Comment by jacobwinch** [comment]
> Then in a following PR we will need to fix the post deploy test for Stripe, which will be broken from #784. I will put a card in the health backlog for that.

My two cents on this @svillafe @joelochlann:

We should not knowingly ship a breaking change to the post-deploy tests until we actually have a PR ready to fix them. (Unless of course something is completely broken and we need to take evasive action e.g. turning off a payment method). 

All engineers working on this project rely on these tests to confidently ship changes. In the past fortnight the tests have caught two significant production bugs and we should be extremely reluctant to allow their failure to become expected or routine.

**Pull request closed without merging**